src/data/mv_polynomial/comm_ring.lean

```diff
@@ -66,6 +66,9 @@ variables (σ a a')
 @[simp] lemma coeff_sub (m : σ →₀ ℕ) (p q : mv_polynomial σ R) :
   coeff m (p - q) = coeff m p - coeff m q := finsupp.sub_apply _ _ _
 
+@[simp] lemma monomial_neg (m : σ →₀ ℕ) (a : R) : monomial m (-a) = -(monomial m a) :=
+by rw [eq_neg_iff_add_eq_zero, monomial_add, neg_add_self, monomial, finsupp.single_zero]
```

Comment:
Can you add `monomial_sub` too? Also, I think this lemma is backwards when compared to `monomial_add`. If so, I'd be inclined to flip it to:
```suggestion
@[simp] lemma monomial_neg (m : σ →₀ ℕ) (a : R) : -(monomial m a) = monomial m (-a) :=
by rw [eq_neg_iff_add_eq_zero, monomial_add, neg_add_self, monomial, finsupp.single_zero]

```
I think you might be able to simplify the proof too, based on how `monomial_add` is proved:
https://github.com/leanprover-community/mathlib/blob/b765d4e7ef1890066ff547fe3b9a2754a19fa537/src/data/mv_polynomial/basic.lean#L210-L212

Reply:
It's concerning that monomial_add and single_add are both simp lemmas, but that the symm of one is used to prove the other.

Reply:
Right, but the proof is doing unfolding that simp won't do, so this doesn't create a cycle.

What's more alarming is that `polynomial.monomial_add` goes in the opposite direction to `mv_polynomial.monomial_add`; so at some point we should proably flip them around.

Reply:
Honestly, I don't even understand why single_add.symm even works for proving monomial_add without rewriting the `monomial`s to `single`s. One would assume that
```
@[simp] lemma monomial_sub (m : σ →₀ ℕ) (a b : R) : monomial m a - monomial m b = monomial m (a - b) := by single_sub.symm
```
Should work, but this throws a type mismatch.

Reply:
What's the error it gives?

Reply:
Here is what my screen displays
<img width="2672" alt="Screen Shot 2021-05-05 at 4 54 34 PM" src="https://user-images.githubusercontent.com/8030599/117208008-a1846080-adc2-11eb-8d7f-447956a756f9.png">

Reply:
In fact, if I go to finsupp/basic.lean I now get an error on ever single lemma there telling me "excessive memory consumption detected at 'replace' (potential solution: increase memory consumption threshold)"

Reply:
Ok, after restarting lean, the issues in finsupp/basic go away, but I still have the type mismatch error